Libraries/Components/CheckBox/CheckBox.android.js

```diff
@@ -9,21 +9,76 @@
  */
 'use strict';
 
-const DeprecatedViewPropTypes = require('DeprecatedViewPropTypes');
-const NativeMethodsMixin = require('NativeMethodsMixin');
-const PropTypes = require('prop-types');
 const React = require('React');
 const StyleSheet = require('StyleSheet');
 
-const createReactClass = require('create-react-class');
 const requireNativeComponent = require('requireNativeComponent');
+const nullthrows = require('nullthrows');
 
-const RCTCheckBox = requireNativeComponent('AndroidCheckBox');
+import type {ViewProps} from 'ViewPropTypes';
+import type {SyntheticEvent} from 'CoreEventTypes';
+import type {NativeComponent} from 'ReactNative';
 
-type DefaultProps = {
-  value: boolean,
-  disabled: boolean,
-};
+type CheckBoxEvent = SyntheticEvent<
+  $ReadOnly<{|
+    target: number,
+    value: boolean,
+  |}>,
+>;
+
+type CommonProps = $ReadOnly<{|
+  /**
+   * Used in case the props change removes the component.
+   */
+  onChange?: ?(event: CheckBoxEvent) => mixed,
+
+  /**
+   * Invoked with the new value when the value changes.
+   */
+  onValueChange?: ?(value: boolean) => mixed,
+
+  /**
+   * Used to locate this view in end-to-end tests.
+   */
+  testID?: ?string,
+
+  /**
+   * Used to get the ref for the native checkbox
+   */
+  // $FlowFixMe: Properly type the ref prop type
+  ref?: $FlowIssue,
+|}>;
+
+type Props = $ReadOnly<{|
+  ...ViewProps,
+  ...CommonProps,
```

Comment:
nit: `ViewProps` are also common props. So, shouldn't we spread `ViewProps` into `CommonProps`?

```JavaScript
type Props = $ReadOnly<{|
  ...CommonProps
  // ...
```

Reply:
👍